graph_analysis/graph_feature_extraction.py

```diff
@@ -0,0 +1,104 @@
+"""This module extracts features in pandas DataFrame format for creating graph structure charts."""
+
+import networkx as nx
+import pandas as pd
+from scipy import stats
+import numpy as np
+
+
+class GraphFeatureExtraction:
+    """
+        A class to extract features from a networkx graph for A->B analysis.
+
+        ...
+
+        Methods:
+            get_cycle_stats()
+                Finds all cycles in a graph and returns stats and dataframe with node count in graph.
+
+            remove_outliers(df, distributions_max=4)
+                Removes outliers from distribution that are more than n standard deviations away from the mean.
+
+            node_distribution_stats()
+                Generates dataframes with paths from A->B for a line chart.
+    """
+    def __init__(self, G):
```

Comment:
consider naming this graph? or is G a standard when working with networkx graphs?